audit-checklists/Arbitrum.md

```diff
@@ -0,0 +1,95 @@
+# Arbitrum security checklist
+
+## General
+
+### Block number
+
+Invoking `block.number` in a smart contract on Arbitrum will return **L1** block number at which sequencer received transaction, not an **L2** block number. Also worth noting, block number being returned is not necessarily the latest L1 block number. It is the latest synced block number and syncing process occurs approximately every minute. In that period there can be ~5 new L1 blocks. So values returned by `block.number` on Arbitrum do not increase continuously but in "jumps" of ~5 blocks.
+
+> Always check for incorrect use of `block.number` on L2s, especially if it is used to track time over short periods.
```

Comment:
- Do you know any other chains that exhibit the same behavior with regards to `block.number` or this only applies to Arbitrum?

Reply:
I'm not aware of this behavior besides Arbitrum (and chains deployed using Orbit - Arbitrum's Nitro stack). Calling block.number on OP stack chains (Optimism, Base, Blast, etc) and some other L2s will return the L2 block number. I updated the 'tip' comment to make it clearer it's in the context of Arbitrum chains specifically